### PR TITLE
Flask service

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# FLASK SERVICE

Expose datastore-version-manager as a flask api. The main purpose of this PR is to restructure the app as a flask api, but did the base line implementation for existing cases with appropriate tests. The further implementation of use cases are best suited for future PRs.

### DONE
* Basic app setup exposing the service with mounted blueprint
* Added logging config from pseudonym-service that follows the kibana model
* Restructured project somewhat to fit the new api setup
* Restructured and reconfigured existing tests
* Added new tests for api
* Updated dockerfile

### ENDPOINTS
Not all endpoints are implemented, but they are all set up in the api module:
* **[GET] /pending-operations**: Get pending operations from datastore
* **[POST] /pending-operations**: Post a new pending operation
* **[PUT] /pending-operations/<dataset_name>**: Update a pending operation with new release status
* **[DELETE] /pending-operations/<dataset_name>**: Delete a pending operation from the datastore
* **[GET] /datastore/bump**: Get a bump manifesto for the datastore
* **[POST] /datastore/bump**:  Request to bump datastore with a posted bump manifesto

### GOING FORWARD
* We should discuss the pending operations set status functionality. It can currently also change the "operation"-field. I am not sure this is necessary. After this is discussed the update pending operation endpoint can be implemented.
* Implement the rest of the versioning use-cases, and simultaneously implementing the api endpoints

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -45,14 +45,13 @@ FROM python:3.9-slim-bullseye
...
 
 RUN pip install -r requirements.txt
 
 #the output is sent straight to terminal without being first buffered
 ENV PYTHONUNBUFFERED 1
 
-ENV PYTHONPATH "/app/datastore_version_manager/:$PYTHONPATH"
-
-ENTRYPOINT [ "python", "datastore_version_manager/commands.py" ]
+CMD [ "gunicorn", "datastore_version_manager.app:app", "--bind", "0.0.0.0:8000", "--workers", "1"]
```
Comment (pawbu) [inline Dockerfile:56]: You can consider setting the timeout for workers. The default is 5 seconds. Maybe a big bump could take more than that?
In such case the syntax is `"--timeout", "20"`

File: datastore_version_manager/api/command_api.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+def get_pending_operations():
+    logger.info('GET /pending-operations')
+    return jsonify(pending_operations.get_datastructure_updates())
+
+
+@command_api.route('/pending-operations', methods=['POST'])
+@validate()
+def add_pending_operation(body: NewPendingOperationRequest):
+    logger.info(f'POST /pending-operations with body {body}')
+    print(f'POST /pending-operations with body {body}')
```
Comment (pawbu) [inline datastore_version_manager/api/command_api.py:33]: print not needed as the msg is logged on line 32

File: datastore_version_manager/api/request_models.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from pydantic import BaseModel, Extra
+
+
+class NewPendingOperationRequest(BaseModel, extra=Extra.forbid):
```
Comment (pawbu) [inline datastore_version_manager/api/request_models.py:4]: 💯

File: datastore_version_manager/service/versioning_service.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+from datastore_version_manager.adapter import (
+    datastore, built_datasets
+)
+from datastore_version_manager.adapter.constants import (
+    USER_CHANGEABLE_RELEASE_STATUSES
+)
+from datastore_version_manager.domain import pending_operations, version_bumper
+
+
+def add_new_dataset(dataset_name: str, description: str, overwrite: bool):
```
Comment (vaskalan) [inline datastore_version_manager/service/versioning_service.py:15]: Is "add_new_draft_dataset" a better name for this function?

**Review by pawbu — APPROVED** [review]
Looks good!
Some minor comments

File: datastore_version_manager/domain/metadata_all.py [hunk 5]
```diff
@@ -38,6 +38,7 @@ def generate_metadata_all_draft():
 def create_new_version(bumped_data_structures: dict, previous_version: str, new_version: str) -> None:
     if previous_version == "0.0.0":
         # no previous metadata_all__x_x_x.json
+        # TODO: make this more generic
         metadata_all = {
             "dataStore": {
                 "name": "no.ssb.fdb",
```
Comment (vaskalan) [inline datastore_version_manager/domain/metadata_all.py:44]: Datastore info hardcoded?

**Review by vaskalan — APPROVED** [review]
Good job!

**Pull request merged**